docs/3-v6-beta/4-flash-fill.md

```diff
@@ -1,9 +1,321 @@
 ---
 sidebar_label: Flash Fill
-description: Coming Soon
-draft: true
+description: Integrate Jupiter Swap via Flash-Fill
 
 ---
 
-# Flash Fill
-Coming Soon
\ No newline at end of file
+# Jupiter Swap via Flash-Fill
+
+Integrate your program with Jupiter Swap without the limitations of CPI via the "Flash-Fill" approach.
+
+Flash-Fill is one of two approaches to integrate Jupiter swap with your protocol. The other approach is [CPI](/docs/v6-beta/cpi-example).
+
+Our team engineered "flash-fill" to allow developers and integrators to utilize the full potential of Jupiter swap with their programs.
+
+:::info Why Flash-Fill?
+_In order to get the best prices and maximum returned amount for any swaps, Jupiter splits and routes an order across multiple dexes in a single transaction to minimize price impact while prioritizing routes with the lowest prices. This results in a single transaction that may involve too many accounts. Transactions on Solana has a maximum of 1232 bytes (as of August 2023) and each account would take up 32 bytes. On top of that, instruction data would also use up precious bytes. As such, the CPI approach would generally fail._
+
+_Although there is a way to force Jupiter's quote API to limit the number of accounts or simply construct a swap instruction that utilizes a single dex, in doing so, you would sacrifice the best routes - not ideal, hence, flash-fill!_
```

Comment:
we should point to `maxAccounts` here.

Reply:
https://github.com/jup-ag/space-station/commit/69255e476a80e5141950e04e1bff475bd779f409